### PR TITLE
Fixes to post media

### PR DESCRIPTION
**Pull request by pfrazee**
A bundle of updates and fixes in this PR:

- Fix a typo (lmao)
- Fix image processing in post embeds
- Add proper validation of mimetype globs in the blob handler
- Modify the resize mode for embed images
- Make the thumbnail optional in external embeds

### CODE REVIEW

File: packages/pds/src/api/app/bsky/util/embeds.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ export const embedsForPosts = async (
   const imgEmbeds = images.reduce((acc, cur) => {
     if (!acc[cur.postUri]) {
       acc[cur.postUri] = {
-        $type: 'app.bksy.embed.images#presented',
```
Comment (dholms) [inline packages/pds/src/api/app/bsky/util/embeds.ts:32]: 🙃

**Review by dholms — APPROVED** [review]
i've made this mistake so many times & it finally slipped into a PR. sorry yall 😅

File: packages/pds/tests/crud.test.ts [hunk 2]
```diff
@@ -156,6 +162,52 @@ describe('crud operations', () => {
...
+    )
+    // Associate image with post, image should be placed in blobstore
+    const res = await aliceClient.app.bsky.feed.post.create(
+      { did: alice.did },
+      {
+        $type: 'app.bsky.feed.post',
+        text: "Here's a key!",
+        createdAt: new Date().toISOString(),
+        embed: {
+          $type: 'app.bksy.embed.images',
```
Comment (devinivy) [inline packages/pds/tests/crud.test.ts:186]:
```suggestion
          $type: 'app.bsky.embed.images',
```

File: packages/pds/src/db/tables/post-embed-external.ts [hunk 3]
```diff
@@ -5,7 +5,7 @@ export interface PostEmbedExternal {
   uri: string
   title: string
   description: string
-  thumbCid: string
+  thumbCid?: string
```
Comment (dholms) [inline packages/pds/src/db/tables/post-embed-external.ts:8]: real quick: this should be `string | null`. kysely returns as `null` not `undefined`

**Review by devinivy — APPROVED** [review]
Big yup!

**Pull request merged**